src/groups/group.jl

```diff
@@ -554,6 +554,215 @@ function inverse_translate_diff!(
+end
+
+@doc doc"""
+    group_log(G::AbstractGroupManifold, y)
+
+Compute the group logarithm of the group element `y`.
+
+Given an element $y ∈ G$, compute the right inverse of the exponential map
+[`group_exp`](@ref), that is, the element $v ∈ 𝔤 = T_e G$, such that $y = \exp_e v$
+
+    group_log(G::AbstractGroupManifold{AdditionOperation}, y)
+
+Compute $v = y$.
+
+    group_log(G::AbstractGroupManifold{MultiplicationOperation}, y)
+
+For number and `AbstractMatrix` types of `y`, compute the usual numeric/matrix logarithm.
+"""
+group_log(M::Manifold, y) = group_log(M, y, is_decorator_manifold(M))
+group_log(M::Manifold, y, ::Val{true}) = group_log(M.manifold, y)
+function group_log(M::Manifold, y, ::Val{false})
+    return error("group_log not implemented on $(typeof(M)) for element $(typeof(y)).")
+end
+function group_log(G::AbstractGroupManifold, y)
+    v = similar_result(G, group_log, y)
+    return group_log!(G, v, y)
+end
+
+group_log!(M::Manifold, v, y) = group_log!(M, v, y, is_decorator_manifold(M))
+group_log!(M::Manifold, v, y, ::Val{true}) = group_log!(M.manifold, v, y)
+function group_log!(M::Manifold, v, y, ::Val{false})
+    return error("group_log! not implemented on $(typeof(M)) for element $(typeof(y)) and vector $(typeof(v)).")
+end
+
+############################
+# Group-specific Retractions
+############################
+
+"""
+    GroupExponentialRetraction{D<:ActionDirection} <: AbstractRetractionMethod
+
+Retraction using the group exponential "translated" to any point on the manifold.
+
+# Constructor
+
+    GroupExponentialRetraction(conv::ActionDirection = LeftAction())
+"""
+struct GroupExponentialRetraction{D<:ActionDirection} <: AbstractRetractionMethod end
+
+function GroupExponentialRetraction(conv::ActionDirection = LeftAction())
+    return GroupExponentialRetraction{typeof(conv)}()
+end
+
+"""
+    GroupLogarithmicInverseRetraction{D<:ActionDirection} <: AbstractInverseRetractionMethod
+
+Retraction using the group logarithm "translated" to any point on the manifold.
+
+# Constructor
+
+    GroupLogarithmicInverseRetraction(conv::ActionDirection = LeftAction())
+"""
+struct GroupLogarithmicInverseRetraction{D<:ActionDirection} <:
+       AbstractInverseRetractionMethod end
+
+function GroupLogarithmicInverseRetraction(conv::ActionDirection = LeftAction())
+    return GroupLogarithmicInverseRetraction{typeof(conv)}()
+end
+
+direction(::GroupExponentialRetraction{D}) where {D} = D()
+direction(::GroupLogarithmicInverseRetraction{D}) where {D} = D()
+
+@doc doc"""
+    retract(
+        G::AbstractGroupManifold,
+        x,
+        v,
+        method::GroupExponentialRetraction{<:ActionDirection},
+    )
+
+Compute the retraction using the group exponential "translated" to any point on the
+manifold. With a group translation $τ_x$ in a specified direction, the retraction is
+
+````math
+\operatorname{retr}_x = τ_x \circ \exp_e \circ (\mathrm{d}τ_x)_x^{-1}
```

Comment:
We should probably use different symbols for Riemannian and group exponentials?

Reply:
Similarly for logarithmic maps.

Reply:
I'm reluctant to use anything too different, to avoid changing from standard convention. I thought giving it a base `e` would be sufficient, but I see now that one can of course have the Riemannian exponential with base `e`. We could capitalize the map name (e.g. `Exp`, and `Log`) or remove the base point entirely, but ultimately, the context will need to make clear which is meant.

Reply:
OK, it should be clear from the text description of formulas which one is used where.